### PR TITLE
New EContact Ingestion 

### PR DESCRIPTION
**Pull request by leogrim**
@eishay @stkem @raymondng 
Main changes besides refactoring:
- we are now tracking all individual emails as EmailAccounts, whether they belong to some user or not. The objective for that table is to fully replace the UserEmailAddress table existing in Shoebox.
- we now allow duplicate email contacts _across different address books_: we no longer have a uniqueness assumption on (userId, emailAccount), we now have a uniqueness assumption on (abookId, emailAccount)
- we now support updating of existing contacts

Main benefits: 
- great data for the graph
- more data for typeahead (we're no longer dropping names for duplicate email contacts)

I'll work on another pull request with one-time data cleaning / migration methods. Then we can drop "Option" for some of the model fields and add constraints to the database.

### CODE REVIEW

File: kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala [hunk 1]
```diff
@@ -120,17 +89,14 @@ class ContactsUpdater @Inject() (
...
         case Some(abookRawInfo) => {
-          val existingEmails = existingEmailSet(userId, origin, abookInfo)
-
-          log.infoP(s"abookRawInfo=$abookRawInfo existingEmails=$existingEmails")
 
           abookRawInfo.contacts.value.grouped(abookUploadConf.batchSize).foreach { g =>
-            val newContactsBuilder = mutable.ArrayBuilder.make[BasicContact]
             batchNum += 1
-            g.foreach { contact =>
+            val basicContacts = g.flatMap { contact =>
```
Comment (eishay) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala]: one of the reasons I don't like one letter variable names is that its not trivial to understand what they mean

Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala]: True, I have seen this in several places but haven't meant to rewrite everything. It should be fairly straightforward, I'll take a closer look and hunt them down once I have completed the logic / data migration.

Comment (eishay) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala]: thanks

Comment (raymondng) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala]: That's actually my bad. I am quite used to single letter variable names but do not feel strongly about it. 
I can take this one as an action item.

Comment (2is10) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala]: I’m okay with one-letter variable names in tiny scopes, like one line or less. It’s much like `_` in that case. But it’s definitely helpful to use more descriptive names in larger scopes.

Comment (raymondng) [inline kifi-backend/modules/abook/app/com/keepit/abook/ABookImporterActor.scala]: Sounds like a good rule of thumb. Thanks Jared.

**Comment by leogrim** [comment]
If I can ever resolve the merge conflicts on this, I will merge :P

**Comment by leogrim** [comment]
Yes! I'm going to work on merging my other shorter pull request. Feel free to comment on this, otherwise I'll apply the db migrations and merge it after lunch :)

**Pull request merged**